packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -51,13 +54,33 @@ export default function createVitePluginAstroServer({
 			const localStorage = new AsyncLocalStorage();
 
 			/** rebuild the route cache + manifest */
-			async function rebuildManifest() {
+			async function rebuildManifest(path: string | null = null) {
 				pipeline.clearRouteCache();
-				routeManifest = injectDefaultDevRoutes(
-					settings,
-					devSSRManifest,
-					await createRouteManifest({ settings, fsMod }, logger), // TODO: Handle partial updates to the manifest
-				);
+
+				// If a route changes, we check if it's part of the manifest and check for its prerender value
+				if (path !== null) {
+					const route = routeManifest.routes.find((r) => path.endsWith(r.component));
+					if (!route) {
+						return;
+					}
+					if (route.type !== 'page' && route.type !== 'endpoint') return;
+
+					const routePath = fileURLToPath(new URL(route.component, settings.config.root));
+					if (!fsMod.existsSync(routePath)) {
+						// Route was renamed so it does not exist anymore
+						return;
```

Comment:
`route.component` can also be an import path (like `@astrojs/starlight/inded.astro`). In that case the `new URL` won't resolve to the file path so it will reach this conditional.

I don't think this is a problem since this would only be reached on a conflict from my other comment. But maybe it should be in the comment.

Reply:
Going by Ema's comment previously it seems import paths are resolved to file paths before this. In that case you can ignore this.